openedx/core/djangoapps/theming/helpers.py

```diff
@@ -52,3 +71,341 @@ def get_themed_template_path(relative_path, default_path, **kwargs):
     if is_stanford_theming_enabled and not is_microsite:
         return relative_path
     return microsite.get_template_path(default_path, **kwargs)
+
+
+def get_template_path_with_theme(relative_path):
+    """
+    Returns template path in current site's theme if it finds one there otherwise returns same path.
+
+    Example:
+        >> get_template_path_with_theme('header')
+        '/red-theme/lms/templates/header.html'
+
+    Parameters:
+        relative_path (str): template's path relative to the templates directory e.g. 'footer.html'
+
+    Returns:
+        (str): template path in current site's theme
+    """
+    site_theme_dir = get_current_site_theme_dir()
+    if not site_theme_dir:
+        return relative_path
+
+    base_theme_dir = get_base_theme_dir()
+    root_name = get_project_root_name()
+    template_path = "/".join([
+        base_theme_dir,
+        site_theme_dir,
+        root_name,
+        "templates"
+    ])
+
+    # strip `/` if present at the start of relative_path
+    template_name = re.sub(r'^/+', '', relative_path)
+    search_path = os.path.join(template_path, template_name)
+    if os.path.isfile(search_path):
+        path = '/{site_theme_dir}/{root_name}/templates/{template_name}'.format(
+            site_theme_dir=site_theme_dir,
+            root_name=root_name,
+            template_name=template_name,
+        )
+        return path
+    else:
+        return relative_path
+
+
+def get_current_theme_template_dirs():
+    """
+    Returns template directories for the current theme.
+
+    Example:
+        >> get_current_theme_template_dirs('header.html')
+        ['/edx/app/edxapp/edx-platform/themes/red-theme/lms/templates/', ]
+
+    Returns:
+        (list): list of directories containing theme templates.
+    """
+    site_theme_dir = get_current_site_theme_dir()
+    if not site_theme_dir:
+        return None
+
+    base_theme_dir = get_base_theme_dir()
+    root_name = get_project_root_name()
+    template_path = "/".join([
+        base_theme_dir,
+        site_theme_dir,
+        root_name,
+        "templates"
+    ])
+
+    return [template_path]
+
+
+def strip_site_theme_templates_path(uri):
+    """
+    Remove site template theme path from the uri.
+
+    Example:
+        >> strip_site_theme_templates_path('/red-theme/lms/templates/header.html')
+        'header.html'
+
+    Arguments:
+        uri (str): template path from which to remove site theme path. e.g. '/red-theme/lms/templates/header.html'
+
+    Returns:
+        (str): template path with site theme path removed.
+    """
+    site_theme_dir = get_current_site_theme_dir()
+    if not site_theme_dir:
+        return uri
+
+    root_name = get_project_root_name()
+    templates_path = "/".join([
+        site_theme_dir,
+        root_name,
+        "templates"
+    ])
+
+    uri = re.sub(r'^/*' + templates_path + '/*', '', uri)
+    return uri
+
+
+def get_current_site():
```

Comment:
yes, `edx-django-sites-extensions ` adds current site to request object, this utility function is used at places where we need current site object but do not have request object like [here](https://github.com/edx/edx-platform/pull/11885/files#diff-a130e1e5dc2d38a357b25f570ab78368R115). This function uses `edx-django-sites-extensions ` to get the current site, [here](https://github.com/edx/edx-platform/pull/11885/files#diff-395f8f21f3f6cb80d7f9c532b06459f5R189) is the code.